modules/programs/vim.nix

```diff
@@ -65,6 +65,18 @@ in {
     programs.vim = {
       enable = mkEnableOption "Vim";
 
+      sourcePackage = mkOption {
+        type = with types; package;
+        default = pkgs.vim_configurable;
+        example = literalExample
+          "pkgs.vim_configurable.override { python = pkgs.python3; }";
+        description = ''
+          A Vim package which will be installed and configured by Home Manager.
+          This can either be one of the default Vim packages from Nixpkgs or a customized
+          package.
+        '';
+      };
```

Comment:
I think it is better to call this `package`, while renaming the `package` to `finalPackage` (that is both `readOnly = true;` and `visible = false;` to avoid mistakes) like did in #1726.

Probably @berbiche can give more details.

Reply:
This pattern is also used in other packages (while there is no `sourcePackage` hits):

```
~/Projects/home-manager master
❯ rg finalPackage
modules/programs/neovim.nix
163:      finalPackage = mkOption {
262:    home.packages = [ cfg.finalPackage ];
265:    programs.neovim.finalPackage = pkgs.wrapNeovimUnstable cfg.package

modules/programs/mpv.nix
72:      finalPackage = mkOption {
176:      programs.mpv.finalPackage = mpvPackage;

modules/programs/emacs.nix
58:      finalPackage = mkOption {
70:    home.packages = [ cfg.finalPackage ];
71:    programs.emacs.finalPackage = emacsWithPackages cfg.extraPackages;

modules/services/emacs.nix
56:      default = if emacsCfg.enable then emacsCfg.finalPackage else pkgs.emacs;
58:        if config.programs.emacs.enable then config.programs.emacs.finalPackage

doc/release-notes/rl-2103.adoc
50:resulting derivation. Use the newly introduced `programs.mpv.finalPackage`

~/Projects/home-manager master
❯ rg sourcePackage
```

Reply:
Yes, naming the option `package` would be more inline with the rest of Home Manager.
It would also be great to add the `finalPackage` option too while at it.